main/src/ca/uwaterloo/flix/language/phase/Lexer.scala

```diff
@@ -39,6 +39,9 @@ import scala.util.Random
   */
 object Lexer {
 
+  /** An average token length estimation used for optimization purposes. */
```

Comment:
Explain better what it is and how it is used.

Also, I would underestimate it.

Reply:
I gave it a go